tests: spi: siwx91x: spi driver fix

smalae opened the pull request:
1. Validate word size to only allow 8-bit and 16-bit.
2. Update CONFIG_SPI_IDEAL_TRANSFER_DURATION_SCALING value for the testcase to pass for higher transfer speeds.

jerome-pouiller reviewed (commented):
I feel the two changes are independant. Maybe it should be two commits?

smalae committed:
drivers: spi: siwx91x: spi driver fix

Validate word size to only allow 8-bit and 16-bit.

Signed-off-by: Sai Santhosh Malae <Santhosh.Malae@silabs.com>

smalae committed:
tests: spi: siwx91x: board config file update

Update CONFIG_SPI_IDEAL_TRANSFER_DURATION_SCALING
value for testcase to pass for higher transfer speeds.

Signed-off-by: Sai Santhosh Malae <Santhosh.Malae@silabs.com>
